protocols/identify/src/behaviour.rs

```diff
@@ -472,6 +472,7 @@ impl PeerCache {
             Some(cache) => cache,
         };
 
+        let addresses = addresses.filter_map(|a| libp2p_swarm::p2p_addr(Some(peer), a).ok());
```

Comment:
We are not dropping them, we are adding `/p2p` suffix if it isn't there! What is confusing is that we are using the `libp2p-swarm` `p2p_addr` method which is fallible when in reality, this can't actually fail. See my comment above :)